rest/src/main/java/com/redhat/ipaas/rest/v1/handler/integration/IntegrationHandler.java

```diff
@@ -92,11 +92,19 @@ public Integration create(Integration integration) {
         return Creator.super.create(updatedIntegration);
     }
 
-    private void updateStatus(String id, Integration.Status desiredStatus) {
+    private void updateDesiredStatus(String id, Integration.Status desiredStatus) {
         this.update(id, new Integration.Builder()
             .createFrom(get(id))
             .token(Tokens.getAuthenticationToken())
             .desiredStatus(desiredStatus)
+            // Set the current status to 'pending' immediately when
```

Comment:
Thanks for the comment - again this is fine for now, but we need a clearer separation of concerns going forward.